libirods_rule_engine_plugin-audit_amqp.cpp

```diff
@@ -141,36 +145,55 @@ irods::error start(irods::default_re_ctx& _u,const std::string& _instance_name)
     messenger = pn_messenger(NULL);
     pn_messenger_start(messenger);
     pn_messenger_set_blocking(messenger, false);  // do not block
-    
-    json_t* obj = json_object();
-    if( !obj ) {
-        return ERROR(SYS_MALLOC_ERR, "json_object() failed");
-    }
+
+    nlohmann::json json_obj;
 
     struct timeval tv;
     gettimeofday(&tv, NULL);
     unsigned long time_ms = tv.tv_sec * 1000 + tv.tv_usec / 1000;
-    std::stringstream time_str; time_str << time_ms;
-    json_object_set(obj, "time_stamp", json_string(time_str.str().c_str()));
 
     char host_name[MAX_NAME_LEN];
     gethostname( host_name, MAX_NAME_LEN );
-    json_object_set(obj, "hostname", json_string(host_name));
 
     pid_t pid = getpid();
-    std::stringstream pid_str; pid_str << pid;
-    json_object_set(obj, "pid", json_string(pid_str.str().c_str()));
 
-    json_object_set(obj, "action", json_string("START"));
+    std::string log_file = str(boost::format("%s/%06i.txt") % log_path_prefix % pid);
+    std::string tmp_msg;
+    try {
+        json_obj["time_stamp"] = std::to_string(time_ms);
+        json_obj["hostname"] = host_name;
+        json_obj["pid"] = std::to_string(pid);
+        json_obj["action"] = "START";
 
-    std::string log_file;
-    if (test_mode) {
-        log_file = str(boost::format("%s/%06i.txt") % log_path_prefix % pid);
-        json_object_set(obj, "log_file", json_string(log_file.c_str()));
+        if (test_mode) {
+            json_obj["log_file"] = log_file;
+        }
+
+        tmp_msg = json_obj.dump(
+            -1,
+            ' ',
+            false,
+            nlohmann::json::error_handler_t::replace
+        );
```

Comment:
This form of `dump()` feels like it shouldn't be used here given that the values being stored in `json_obj` are strings and integers. `hostname` and `log_file` are probably the only properties that can contain invalid UTF-8 characters and that doesn't feel like something the audit plugin should be concerned with.

Do we expect invalid UTF-8 characters to appear here? (seems unlikely)
Should invalid UTF-8 characters be replaced or result in an exception?